Add "Adding a Provider" section to README (#6)
calibreapp/react-live-chat-loader: README.md

  `2000`. Set to `0` to never load. This value is used in a `setTimeout` in
  browsers that don't support `requestIdleCallback`.

## Supported Providers

> Currently there are two supported providers:

Reviewer: I was wondering if we could use a `<details>` expander to list the provided examples? Then we could remove the "Currently there are two supported providers" label and shorten up the README a bunch. What do you think?

Author: 👍 Good call. Updated Supported Providers to use `<details>` and put some templates for provider and component in `<details>` too.

You can customise the color of the Intercom widget by passing a `color` prop to
the `Intercom` component.

## Adding a provider

> To add a new provider you need to add the following:

Reviewer: ```suggestion
To contribute a new provider, follow these steps:
```

#### 1. Create provider file

Create a new provider file at `src/providers/providerName.js` which exports the
following:

- `domain`: A string of the domain where the provider script is loaded from
  that will be used in a `preconnect` link.
- `load`: Function which when called will load and initialize the provider
  script. It should accept props and use the `providerKey` as the `app_id` or
  `api_key`. For consistency, it should call a `loadScript` function.
- `open`: Function which when called will open the provider chat.

> - `close`: Function which when called will close the provider chat.

Reviewer: It'd be great to have an example immediately above this for easy copy-paste

> Import that new file in `src/providers/index.js` and add it to `Providers`.

Reviewer: ```suggestion
Import the new file in `src/providers/index.js` and add it to `Providers`.
```

The name of this file will be the `providerKey` used in the
`LiveChatLoaderProvider`.

#### 2. Create component

Create a new component in `src/Components/ProviderName/index.js` which
replicates the chat widget. Do not worry about loading animations as the widget

> will be shown instantly on page load.

Reviewer: Should we say something like "your fake chat widget should be positioned using a z-index so that it sits immediately above the chat widget that is being replaced?"

Export the component from `src/index.js`

#### 3. Update README

> Add your new provider to this README.

Reviewer: And maybe add yourself to the contributors section? 

(I guess we'd need a contributors section?)

Author: Let's push contributors to another PR.
